src/blog/2024/01/send-a-file.md

```diff
@@ -0,0 +1,162 @@
+<!--more-->
+
+# Why would you need to send a file to Node-RED?
+
+Often times it is necessary to update lookup tables in a SQL database, but you don't necessarily want to give access to everyone to edit the database, nor do you want to have to do it all yourself. This can often be seen when new products are introduced into a manufacturing facility. It may not be often, but enough that it warrants its own application. This process will guide you in a way that will enable your teammates to upload the files to the system themselves.
+
+# Summary
+
+There are many approaches that can be taken when solving this. We are going to go over 4. The first is the most simple approach but very manual.  A simple cURL command to send the file.  The second is to leverage FlowFuse's Dashboards to allow the user to navigate to the webpage upload a file and visualize before sending it on to the next process.  The third approach is a simple script that will be shared below. It is a simple Python application that allows the user to send a file with a simple command, but this might require a little more technical skills that the end user may not feel comfortable with. Which brought us to last option. A web-based application that allows the user to upload files to a browser with a selectable endpoint. All have their pros and cons. I will let you decide which is best for your application.
+
+
+## cURL
+
+Honestly it doesn't get much more simple than this, but it is probably not for everyone.  Not everyone has access to a device that has cURL available.  Generally it is only available on Linux and UNIX.  It was announced to be available in Windows, but it isn't a direct translation and function more like a alias to "[Invoke-WebRequest](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-webrequest?view=powershell-7.4)".  It isn't a one to one.  None the less here is an example cURL command that will work with this [flow](#node-red-ingress).
+
+```bash
+curl -F 'data=@/path/to/file.csv' http://localhost:1880/fileupload
+```
+
+
+## NodeRED Dashboard (FlowFuse)
+
+This simple flow allows the user to visualize data from a CSV in FlowFuse Dashboards. The button then allows the user to initiate a request to send the data to the next step. This next step could be anything from loading into a SQL Database to saving it locally.
+
+![csv dashboard](./images/csv-dashboard.png)
+
+<iframe width="100%" height="225px" src="https://flows.nodered.org/flow/8c505039ac1b8dbed2bee1e22ee2975a/share?height=100" allow="clipboard-read; clipboard-write" style="border: none;"></iframe>
+
+A link to the flow can be found [here](https://flows.nodered.org/flow/8c505039ac1b8dbed2bee1e22ee2975a).
+
+To import the flow, follows these [instructions](https://flowfuse.com/blog/2023/03/3-quick-node-red-tips-5/#1.-copy-and-share-your-flows-using-export-and-import).
+
+
+## Simple Script
```

Comment:
Honestly I believe we should not include this; why complicate the article? IMO the dashboard method is best and is not explained well. The article should explain better why that works, and how it work. That's it for me, no python, no scripts. Even skip curl.